jdaviz/configs/imviz/plugins/catalogs/catalogs.py

```diff
@@ -324,23 +326,61 @@ def plot_selected_points(self):
                                                            getattr(y, 'value', y))
 
     def vue_zoom_in(self, *args, **kwargs):
-        """This function will zoom into the image based on the selected points"""
+
+        self.zoom_to_selected(return_bounding_box=True)
+
+    def zoom_to_selected(self, padding=0.02, return_bounding_box=False):
+        """
+        Zoom to a region containing the currently selected points in the catalog.
+
+        Parameters
+        ----------
+        padding : float, optional
+            A fractional value representing the padding around the bounding box
+            of the selected points. It is applied as a proportion of the largest
+            dimension of the current extent of loaded data. Defaults to 0.02.
+        return_bounding_box : bool, optional
+            If True, returns the bounding box of the zoomed region as
+            ((x_min, x_max), (y_min, y_max)). Defaults to False.
```

Comment:
other places we just return a single list `(x_min, x_max, y_min, y_max)`, do we want to keep that convention here or is it useful to have the nested tuples?

Reply:
ill change it